presto-main/src/main/java/com/facebook/presto/sql/planner/plan/Assignments.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sql.planner.plan;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.ExpressionRewriter;
+import com.facebook.presto.sql.tree.ExpressionTreeRewriter;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.StreamSupport;
+
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableMap;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class Assignments
+{
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static Assignments identity(Iterable<Symbol> symbols)
+    {
+        return builder().putIdentities(symbols).build();
+    }
+
+    public static Assignments of()
+    {
+        return builder().build();
+    }
+
+    public static Assignments of(Symbol symbol, Expression expression)
+    {
+        return builder().put(symbol, expression).build();
+    }
+
+    public static Assignments of(Symbol symbol1, Expression expression1, Symbol symbol2, Expression expression2)
+    {
+        return builder().put(symbol1, expression1).put(symbol2, expression2).build();
+    }
+
+    private final Map<Symbol, Expression> assignments;
+    private final List<Symbol> outputs;
+
+    @JsonCreator
+    public Assignments(@JsonProperty("assignments") Map<Symbol, Expression> assignments)
+    {
+        this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "assignments is null"));
+        this.outputs = ImmutableList.copyOf(assignments.keySet());
+    }
+
+    public List<Symbol> getOutputs()
+    {
+        return outputs;
+    }
+
+    public Map<Symbol, Expression> getAssignments()
```

Comment:
I renamed that to `getMap` as there is no conversion (it just a getter) so prefix `to` or `as` could be misleading